src/Provider/PrototypeQueueProvider.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Queue\Provider;
+
+use Yiisoft\Queue\QueueInterface;
+
+/**
+ * Queue provider that only changes the channel name of the base queue.
+ * It can be useful when your queues used the same adapter.
+ */
+final class PrototypeQueueProvider implements QueueProviderInterface
+{
+    /**
+     * @param QueueInterface $baseQueue Base queue to use for creating queues.
+     */
+    public function __construct(
+        private readonly QueueInterface $baseQueue,
+    ) {
+    }
+
+    public function get(string $channel): QueueInterface
+    {
+        return $this->baseQueue->withChannelName($channel);
```

Comment:
It's not enough just to set a channel name into a queue object. In the old factory we used a default (base) adapter too:
```php
return $this->queue->withChannelName($channel)->withAdapter($this->defaultAdapter->withChannel($channel));
```

Reply:
I plan fix it another PR. `withChannelName` in `Queue` should be enough and `Queue` should configure own adapter himself.

Reply:
Anyway this PR should work as expected before merge into `master`.

Reply:
Fixed in `Queue::withChannelName`.

Reply:
We decided to remove `Queue::withChannelName` metod in future PR.